pandas/core/indexing.py

```diff
@@ -1302,10 +1303,12 @@ def _validate_read_indexer(
             # code, so we want to avoid warning & then
             # just raising
             if not ax.is_categorical():
+                not_found = list(key[missing_mask])
```

Comment:
instead of doing this you can create an Index which has the natural formatters

Reply:
Done, and I now also use the option context to limit the number of items displayed.

Note, the `if raise_missing` block just above this uses a list to print the missing keys [here](https://github.com/pandas-dev/pandas/blob/7d0ee96f9aedbbcd50781e1b8536f1914ecae984/pandas/core/indexing.py#L1295) Independently of this PR, if raise_missing==False, an error would be raised directly below regardless (unless the axis is categorical). Not sure if this is the expected/desired behaviour? Perhaps one for another ticket/PR.